### PR TITLE
Adds support for adding users to groups

### PR DESCRIPTION
**Pull request by ax-taustgen**
The jfrog api currently supports updating groups with the list of users that belong to that group. This adds the same functionality to terraform.

Previously was using the atlassian artifactory plugin which is 2 years out of date and lacking features, I updated it to use the new jfrog owned client which is currently maintained.

### CODE REVIEW

**Comment by ax-taustgen** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by ax-taustgen** [comment]
@chb0github Could use advice with the following two questions, 
1. Is it possible to retrigger the github action  without pushing a no-op commit, I don't have high enough permissions on this repo to restart the workflow from the UI

2. How are we supposed to run the test suite, the README mentions a docker-compose file but currently it doesnt exist and the makefile lists test as broken

**Comment by ax-taustgen** [comment]
I've run a make test but since I'm having to replace the client I really would like to run an integration test against a local instance, but it looks like that functionality is currently working

**Comment by chb0github** [comment]
Any updates?

**Comment by chb0github** [comment]
If you can finish this soon, that'd be great. It's next up on my migration path from the atlassian client

**Comment by ax-taustgen** [comment]
@chb0github This is ready for review now, I found a work around for UpdateGroup using POST instead of PUT and preventing disassociating users because of it. 


```
=== RUN   TestAccGroup_basic
--- PASS: TestAccGroup_basic (3.26s)
=== RUN   TestAccGroup_full
--- PASS: TestAccGroup_full (10.85s)
```

File: pkg/artifactory/resource_artifactory_group.go [hunk 1]
```diff
@@ -55,74 +53,97 @@ func resourceArtifactoryGroup() *schema.Resource {
...
-	if resp == nil {
-		return fmt.Errorf("no response returned during resourceGroupRead")
-	}
-	// If we 404 it is likely the resources was externally deleted
-	// If the ID is updated to blank, this tells Terraform the resource no longer exist
-	if resp.StatusCode == http.StatusNotFound {
-		d.SetId("")
-		return nil
-	} else if err != nil {
+	group, err := resourceGroupGet(d, m)
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_group.go:145]: I don't know why the original status code check was there, but I am not comfortable removing it without an explanation

File: pkg/artifactory/resource_artifactory_group_test.go [hunk 2]
```diff
@@ -71,14 +127,13 @@ func testAccCheckGroupDestroy(id string) func(*terraform.State) error {
 		}
 
 		_, resp, err := client.V1.Security.GetGroup(context.Background(), rs.Primary.ID)
+		if resp.StatusCode == http.StatusNotFound {
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_group_test.go:130]: If you don't initially check `err`, then you have no guarantee that `resp` isn't`nil` - So, please check that `resp != nil` - I've had this cause me crashes

File: pkg/artifactory/resource_artifactory_group_test.go [hunk 3]
```diff
@@ -71,14 +127,13 @@ func testAccCheckGroupDestroy(id string) func(*terraform.State) error {
 		}
 
 		_, resp, err := client.V1.Security.GetGroup(context.Background(), rs.Primary.ID)
```
Comment (chb0github) [inline pkg/artifactory/resource_artifactory_group_test.go:129]: You're still using the old atlassian client to test?

**Review by chb0github — CHANGES_REQUESTED** [review]
Please address these few things. This is pretty close

**Comment by ax-taustgen** [comment]
Could use a rereview when you get a minute

**Comment by chb0github** [comment]
I think I am ok with what I see here. However, I have been removing all atlassian references and putting them into an entirely different branch and will release them all as a major version bump. When do you need to see this in a release?

**Review by chb0github — COMMENTED** [review]
I am ok with what I see, but I want to hold off on a release because I want to do a major version bump when all atlassian code is gone.

**Comment by ax-taustgen** [comment]
I would really prefer to get this available, its a blocker for rolling out artifactory to my product teams.  I could maybe wait a sprint, but I've got a cutover coming up soon and really don't want have to manually provision it all.

**Comment by chb0github** [comment]
How about: Come hell-or-highwater, the end of this month 2.3.1 is released?

**Comment by ax-taustgen** [comment]
👍  sooner is better than later but if you can get it done before September, I can probably make it work

**Comment by ax-taustgen** [comment]
Can you approve this so I can merge? Or are there more changes needed still?

**Comment by chb0github** [comment]
I merged it. I will let you know when it releases

**Comment by ax-taustgen** [comment]
Thank you!

**Comment by ax-taustgen** [comment]
Hows 2.3.1 coming?

**Comment by chb0github** [comment]
Going out today. Sorry. I am just seeing this. I got additional crap
dropped in my plate. It's going out with sdk2 and I am exhausted from
looking at this code. But, I have a way clearer understanding of it. Most
of the tests pass - 1 can't pass and another requires implementation of
retry logic.

**Comment by chb0github** [comment]
released

**Pull request merged**